File: Day-1/Two-Sum.c
#include <stdio.h>
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    *returnSize = 2;  // We will return exactly 2 indices
    int* result = (int*)malloc(2 * sizeof(int));
    
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (nums[i] + nums[j] == target) {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }
    
    *returnSize = 0;  // No solution found
    return result;
}

int main() {
    int nums[] = {2, 7, 11, 15};
    int target = 9;
    int returnSize;
    
    int* result = twoSum(nums, 4, target, &returnSize);
    
    if (returnSize == 2) {
        printf("[%d, %d]\n", result[0], result[1]);
    } else {
        printf("No solution found\n");
    }
    
    free(result);  // Don't forget to free the allocated memory
    return 0;
}

File: Day-1/Two-Sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, (len(nums))):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []  # Return empty list if no solution found

def main():
    # Test the solution
    solution = Solution()
    nums = [2, 7, 11, 15]
    target = 9
    result = solution.twoSum(nums, target)
    print(result)  # Add this line to see the output

if __name__ == "__main__":
    main()
